Code/Framework/AzToolsFramework/Tests/EntityInspectorTests.cpp

```diff
@@ -23,11 +23,16 @@
 #include <AzToolsFramework/API/ToolsApplicationAPI.h>
 #include <AzToolsFramework/Entity/EditorEntityContextBus.h>
 #include <AzToolsFramework/Entity/EditorEntityHelpers.h>
+#include <AzToolsFramework/ToolsComponents/EditorInspectorComponent.h>
 #include <AzToolsFramework/UnitTest/ToolsTestApplication.h>
+#include <AzToolsFramework/ToolsComponents/EditorInspectorComponentBus.h>
 
 // Inspector Test Includes
 #include <AzToolsFramework/UI/ComponentPalette/ComponentPaletteUtil.hxx>
 
+#include <gmock/gmock.h>
```

Comment:
include azstd memory

Reply:
But why?  Nobody mentions an allocator anywhere in the cpp file?

Reply:
Because of AZStd::unique_ptr in line 485 if I'm not mistaken

Reply:
I tried the following mini app in azcore, and it had no issues:
```cpp

struct S
{
    int retval = 0;
};

int main([[maybe_unused]]int argc,[[maybe_unused]] char* argv[])
{
    AZStd::unique_ptr<S> s = AZStd::make_unique<S>();

    return s->retval;
}
```


It turns out that, well, unique_ptr includes memory.h anyway.  

I'm in the habit of only including what I actually use in the cpp file, if possible.  I didn't use memory so I didn't include it.  azstd::unique_ptr does use memory, so it includes it.